src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,362 @@
+            std::uint16_t                   minorVersion(Traits<std::uint16_t>::deserialize(ar));
+
+            if(majorVersion != 1 || minorVersion != 0)
+                throw std::runtime_error("Unsupported archive version");
+
+            // Data
+            std::vector<std::double_t> norms(Traits<std::vector<std::double_t>>::deserialize(ar));
+
+            return NormalizeTransformer<InputT>(std::move(norms));
+        }()
+    ) {
+}
+
+template <typename InputT>
+void NormalizeTransformer<InputT>::save(Archive &ar) const /*override*/ {
+    // Version
+    Traits<std::uint16_t>::serialize(ar, 1); // Major
+    Traits<std::uint16_t>::serialize(ar, 0); // Minor
+
+    // Data
+    Traits<std::vector<std::double_t>>::serialize(ar, _norms);
+}
+
+template <typename InputT>
+bool NormalizeTransformer<InputT>::operator==(NormalizeTransformer const &other) const {
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#endif
+
+    return this->_norms == other._norms;
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#endif
+
+}
+
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename InputT>
+void NormalizeTransformer<InputT>::execute_impl(InputT const &input, typename BaseType::CallbackFunction const &callback) /*override*/ {
+    execute_impl(const_cast<IteratorType&>(std::get<0>(input)), std::get<1>(input), callback, std::integral_constant<bool, Microsoft::Featurizer::Traits<ValueType>::IsNullableType>());
+}
+
+template <typename InputT>
+void NormalizeTransformer<InputT>::execute_impl(IteratorType &begin, IteratorType const &end, typename BaseType::CallbackFunction const &callback, std::true_type) {
+    // ----------------------------------------------------------------------
+    using InputTraits                       = Traits<ValueType>;
+    using TransformedTraits                 = Traits<std::double_t>;
+    // ----------------------------------------------------------------------
+    if (std::distance(begin, end) < 0) {
+        throw std::runtime_error("Input iterators to VectorNormsEstimator are invalid!");
+    }
+    if (_row >= _norms.size()) {
+        throw std::runtime_error("Number of norms is not aligned with number of rows!");
+    }
+
+
+    std::vector<std::double_t> res;
+    res.reserve(static_cast<size_t>(std::distance(begin, end)));
+    while (begin != end) {
+        if(InputTraits::IsNull(*begin)) {
+            res.emplace_back(TransformedTraits::CreateNullValue());
+            ++begin;
+            continue;
+        }
+        res.emplace_back(static_cast<std::double_t>(static_cast<long double>(InputTraits::GetNullableValue(*begin)) / static_cast<long double>(_norms[_row])));

```

Comment:
Rather than embedding the continue in the if clause, I'd recommend removing the code that increments the iterator and continue and adding an else before the other emplace. Doing this better encapsulates what the code is doing: adding a value to the results and then moving the iterator to point to the next item.